### PR TITLE
Update rails to 5.2.x

### PR DESCRIPTION
**Pull request by iancanderson**
Closes https://github.com/education/classroom/issues/2095

## What
Update rails to the latest stable version so we can be ready to go to 6.0 when it's stable!

## TODO
- [x] Fix test failures
- [x] Fix deprecation warnings in test suite
- [x] Review changelogs for updated gems to make sure there are no surprising new behaviors/breaking changes for the app
  - [x] rails
  - [x] bullet
  - [x] dotenv-rails
  - [x] lograge
- [X] Add bootsnap (NOTE: Bootsnap was manually added 2 years ago, I added a step in our bootstrap script to clear it's cache as that can become massive over time)
- [x] Locally test - Make sure existing sessions continue to be valid in Rail 5.2.x
- [ ] Manually test branch on production

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -8,7 +8,7 @@ git_source(:github) do |repo_name|
 end
 
 ruby File.read(File.expand_path("../.ruby-version", __FILE__)).chomp
-gem "rails", "~> 5.1", ">= 5.1.7"
+gem "rails", "~> 5.2.3"
```
Comment (d12) [inline Gemfile]: 🚀

File: app/models/roster_entry.rb [hunk 2]
```diff
@@ -13,16 +13,19 @@ class IdentifierCreationError < StandardError; end
...
+      ON roster_entries.user_id = assignment_repos.user_id
+      AND assignment_repos.assignment_id='#{sql_formatted_assignment_id}'
+     SQL
     order("assignment_repos.created_at")
-      .joins <<~SQL
-        LEFT OUTER JOIN assignment_repos
-        ON roster_entries.user_id = assignment_repos.user_id
-        AND assignment_repos.assignment_id='#{sql_formatted_assignment_id}'
-      SQL
+      .joins(join_query)
```
Comment (d12) [inline app/models/roster_entry.rb]: Did you mean to use `Arel.sql` here?

File: config/initializers/new_framework_defaults.rb [hunk 3]
```diff
@@ -20,6 +20,3 @@
 
 # Require `belongs_to` associations by default. Previous versions had false.
 Rails.application.config.active_record.belongs_to_required_by_default = false
-
-# Do not halt callback chains when a callback returns false. Previous versions had true.
-ActiveSupport.halt_callback_chains_on_return_false = true
```
Comment (d12) [inline config/initializers/new_framework_defaults.rb]: Will this break any expectations within the app that returning false within a callback halts the chain?

File: config/environments/production.rb [hunk 4]
```diff
@@ -41,6 +40,9 @@
   # config.action_dispatch.x_sendfile_header = 'X-Sendfile' # for Apache
   # config.action_dispatch.x_sendfile_header = 'X-Accel-Redirect' # for NGINX
 
+  # Store uploaded files on the local file system (see config/storage.yml for options)
+  config.active_storage.service = :local
```
Comment (jeffrafter) [inline config/environments/production.rb]: This feels odd for production? Is this meant to be only for development?

Comment (JessRudder) [reply config/environments/production.rb]: @shaunakpp, I believe you were working with Ian on the upgrade. Do you have insight into why this was added? And if it should be for production? If not, I can dig deeper. Just didn't want to chase answers down if you already had them. 😄

Comment (shaunakpp) [reply config/environments/production.rb]: It came up with the `app:update` task afaik, but we don't really use  active storage in classroom, yet, so we can comment it out I think.

File: config/initializers/new_framework_defaults.rb (hunk 3 above)
Comment (JessRudder) [reply config/initializers/new_framework_defaults.rb]: Thanks for bringing this up. I dug into all of our existing callbacks and they should not be affected by this change.

For current before callbacks, the methods either end with a statement that will always be true (a var assignment of a truthy value, a string, etc), they explicitly have `true` as the final return value to ensure the callback won't be halted or they raise an exception which intentionally halts the callbacks.

**Pull request merged**